utils/deploys/deployUtils.ts

```diff
@@ -525,4 +531,60 @@ export async function updateSetManager(
     });
     await writeTransactionToOutputs(setManagerTransaction.transactionHash, description);
   }
+}
+
+async function getEnvironment(hre: HardhatRuntimeEnvironment) {
+  const [owner] = await getAccounts();
+  const instanceGetter = new InstanceGetter(owner.wallet);
+  const { rawTx } = hre.deployments;
+  const { deployer } = await hre.getNamedAccounts();
+  const networkConstant = await getNetworkConstant();
+
+  return {
+    instanceGetter,
+    rawTx,
+    deployer,
+    networkConstant,
+  };
+}
+
+export async function enableModuleOnSetController(
+  moduleName: string,
+  hre: HardhatRuntimeEnvironment,
+  skipProductionCheck: boolean = false,
+): Promise<void> {
+  const {
+    instanceGetter,
+    rawTx,
+    deployer,
+    networkConstant,
+  } = await getEnvironment(hre);
+
+  const moduleAddress = await getContractAddress(moduleName);
+  const controllerAddress = await findDependency(CONTROLLER);
+  const controllerInstance = await instanceGetter.getController(controllerAddress);
+
+  if (!await controllerInstance.isSystemContract(moduleAddress)) {
+    const data = controllerInstance.interface.encodeFunctionData(
+      "addModule",
+      [moduleAddress]
+    );
+    const description = `Add ${moduleName} to Controller`;
+
+    if ((networkConstant === "production" || process.env.TESTING_PRODUCTION) && !skipProductionCheck) {
+      await saveDeferredTransactionData({
+        data,
+        description,
+        contractName: CONTROLLER,
+      });
+    } else {
+      const addModuleTransaction: any = await rawTx({
+        from: deployer,
```

Comment:
Here I am trying to replicate functionality in [set-v2-deployments](https://github.com/SetProtocol/set-v2-deployments/blob/599859a9dff846dc933253f7fe3f27e6e7d41c27/utils/deployUtils.ts#L132)

I get an error for `Ownable: caller is not the owner` because of the `deployer` grabbed above differs from the one in set-v2-deployments?